Bug 1894889: Update query

rebeccaalpert opened the pull request:
Added updated node query to reflect desired filesystem exclusions.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1894889.

CC @openshift/openshift-team-monitoring.

palonsoro commented:
Hi,

Not very expert in how console code works, but doing a `grep -r` I find what look like more points to revise in regards this bug, can you please have a look at them?

```
$ grep -r node_filesystem_size_bytes
frontend/packages/console-app/src/components/nodes/NodesPage.tsx:        'sum by (instance) (node_filesystem_size_bytes{fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"} - node_filesystem_free_bytes{fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"})',
frontend/packages/console-app/src/components/nodes/NodesPage.tsx:        'sum by (instance) (node_filesystem_size_bytes{fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"})',
frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts:    `sum(node_filesystem_size_bytes{instance="<%= node %>",fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"} - node_filesystem_avail_bytes{instance="<%= node %>",fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"})`,
frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts:    `sum(node_filesystem_size_bytes{instance='<%= node %>',fstype!~"tmpfs|squashfs",mountpoint!~"/usr|/var"})`,
frontend/packages/metal3-plugin/src/components/baremetal-hosts/dashboard/queries.ts:  [HostQuery.STORAGE_TOTAL]: _.template(`sum(node_filesystem_size_bytes{instance=~'<%= node %>'})`),
frontend/public/components/dashboard/dashboards-page/cluster-dashboard/queries.ts:    '(sum(node_filesystem_size_bytes) - sum(node_filesystem_free_bytes))',
frontend/public/components/dashboard/dashboards-page/cluster-dashboard/queries.ts:  [OverviewQuery.STORAGE_TOTAL]: 'sum(node_filesystem_size_bytes)',
```

palonsoro commented:
In general, what we would need is:
- We need `nsfs` filesystems to be skipped, if any
- We need `/etc` and `/sysroot` mount points to be skipped, if any

rebeccaalpert commented:
Thanks @palonsoro, I'll take a look!

rebeccaalpert commented:
@palonsoro - Can you confirm that this is what you're looking for? I updated the additional queries (`frontend/public/components/dashboard/dashboards-page/cluster-dashboard/queries.ts` no longer exists and was replaced with one of the other files I updated).

palonsoro commented:
I think so, but my understanding on the console code is very limited. You just need to ensure that the correct filters are in place wherever the storage of a node is calculated.

Thanks and regards.

paulfantom commented on frontend/packages/console-shared/src/promql/cluster-dashboard.ts:
```diff
@@ -44,7 +44,7 @@ const top25Queries = {
     'topk(25, sort_desc(node_memory_MemTotal_bytes - node_memory_MemAvailable_bytes))',
   [OverviewQuery.NODES_BY_STORAGE]:
-    'topk(25, sort_desc(sum(node_filesystem_size_bytes{fstype!="tmpfs",mountpoint!="/boot|/boot/efi"} - node_filesystem_avail_bytes{instance=~".*",fstype!="tmpfs",mountpoint!="/boot|/boot/efi"}) BY (instance)))',
+    'topk(25, sort_desc(sum(node_filesystem_size_bytes{fstype!="tmpfs|squashfs|nsfs",mountpoint!="/boot|/boot/efi|/usr|/var|/etc|/sysroot"} - node_filesystem_avail_bytes{instance=~".*",fstype!="tmpfs|squashfs|nsfs",mountpoint!="/boot|/boot/efi|/usr|/var|/etc|/sysroot"}) BY (instance)))',
```
Is it intended to exclude `/boot|/boot/efi` only here and not in other queries?

rebeccaalpert replied:
@paulfantom I don't have subject matter expertise on what mountpoints we need to exclude, so I just added on the ones the other had. Do you know who would be able to verify what's required?

spadgett replied:
I'm not sure why we would exclude that in only one place.

@rawagner Do you happen to know the history of this query?

rawagner replied:
I dont remember. I think at the beginning we had only `node_filesystem_size_bytes`, than the exclusions were added by @rebeccaalpert IIRC.

rebeccaalpert replied:
Yup, Monitoring told me to add it initially, but now it's being questioned and I've been told they don't have subject matter expertise. I don't write these queries.

paulfantom commented:
Syntax of promql queries LGTM.

rawagner commented:
There's another PR for filesystem queries https://github.com/openshift/console/pull/7201 . Which is correct ?

spadgett commented:
Closing in favor of #7201
